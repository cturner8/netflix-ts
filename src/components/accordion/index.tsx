import { createContext, useContext, useState } from "react";
import {
  Inner,
  Container,
  Title,
  Frame,
  Item,
  Header,
  Body,
} from "./styles/accordion";

interface Props {}

interface TitleProps {}
interface FrameProps {}
interface ItemProps {}
interface HeaderProps {}
interface BodyProps {}

interface Compound {
  Title: React.FC<TitleProps>;
  Frame: React.FC<FrameProps>;
  Item: React.FC<ItemProps>;
  Header: React.FC<HeaderProps>;
  Body: React.FC<BodyProps>;
}

type ToggleShow = boolean;
type SetToggleShow = React.Dispatch<React.SetStateAction<ToggleShow>>;

interface Context {
  toggleShow?: ToggleShow;
  setToggleShow?: SetToggleShow;
}

const ToggleContext = createContext<Context>({});

const Accordion: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};

const ATitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const AFrame: React.FC<FrameProps> = ({ children, ...restProps }) => {
  return <Frame {...restProps}>{children}</Frame>;
};

const AItem: React.FC<ItemProps> = ({ children, ...restProps }) => {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

const AHeader: React.FC<HeaderProps> = ({ children, ...restProps }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);
  const onClick = () => {
    if (setToggleShow) {
      setToggleShow((prevToggle: ToggleShow) => !prevToggle);
    }
  };
  const actionButton = toggleShow ? (
    <img src="/images/icons/close-slim.png" alt="close" />
  ) : (
    <img src="/images/icons/add.png" alt="open" />
  );

  return (
    <Header onClick={onClick} {...restProps}>
      {children}
      {actionButton}
    </Header>
  );
};

const ABody: React.FC<BodyProps> = ({ children, ...restProps }) => {
  const { toggleShow } = useContext(ToggleContext);

  return toggleShow ? <Body {...restProps}>{children}</Body> : <></>;
};

Accordion.Title = ATitle;
Accordion.Frame = AFrame;
Accordion.Item = AItem;
Accordion.Header = AHeader;
Accordion.Body = ABody;

export { Accordion };
