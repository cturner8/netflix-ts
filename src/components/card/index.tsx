import { createContext, useState, useContext, HTMLAttributes } from "react";
import {
  Container,
  Group,
  Title,
  SubTitle,
  Text,
  Meta,
  Item,
  Image,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
  Content,
  Entities,
} from "./styles/card";

interface Props {}

interface ItemFeature {}

export interface GroupProps {
  flexDirection: string;
  alignItems: string;
  margin: string;
}

interface TitleProps {}
interface SubTitleProps {}
interface TextProps {}
interface MetaProps {}
interface ItemProps extends HTMLAttributes<HTMLDivElement> {
  item: ItemFeature;
}
interface ImageProps {}
interface EntityProps {}
export interface FeatureProps {
  src: string;
}
export interface FeatureTextProps {
  fontWeight: string;
}
export interface MaturityProps {
  rating: number;
}

interface Context {
  showFeature: boolean;
  setShowFeature: Function;
  itemFeature: ItemFeature;
  setItemFeature: Function;
}

export const FeatureContext = createContext<Context>({
  showFeature: false,
  setShowFeature: () => {},
  itemFeature: {},
  setItemFeature: () => {},
});

interface Compound {
  Group: React.FC<GroupProps>;
  Title: React.FC<TitleProps>;
  SubTitle: React.FC<SubTitleProps>;
  Text: React.FC<TextProps>;
  Meta: React.FC<MetaProps>;
  Item: React.FC<ItemProps>;
  Image: React.FC<ImageProps>;
  Entities: React.FC<EntityProps>;
}

const Card: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});

  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...restProps}>{children}</Container>
    </FeatureContext.Provider>
  );
};

const CGroup: React.FC<GroupProps> = ({ children, ...restProps }) => {
  return <Group {...restProps}>{children}</Group>;
};

const CTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const CSubTitle: React.FC<SubTitleProps> = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

const CText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const CMeta: React.FC<MetaProps> = ({ children, ...restProps }) => {
  return <Meta {...restProps}>{children}</Meta>;
};

const CItem: React.FC<ItemProps> = ({ item, children, ...restProps }) => {
  const { setShowFeature, setItemFeature } = useContext(FeatureContext);

  return (
    <Item
      onClick={() => {
        setItemFeature(item);
        setShowFeature(true);
      }}
      {...restProps}
    >
      {children}
    </Item>
  );
};

const CImage: React.FC<ImageProps> = ({ ...restProps }) => {
  return <Image {...restProps} />;
};

const CEntities: React.FC<EntityProps> = ({ children, ...restProps }) => {
  return <Entities {...restProps}>{children}</Entities>;
};

Card.Group = CGroup;
Card.Title = CTitle;
Card.SubTitle = CSubTitle;
Card.Text = CText;
Card.Meta = CMeta;
Card.Item = CItem;
Card.Image = CImage;
Card.Entities = CEntities;

export { Card };
