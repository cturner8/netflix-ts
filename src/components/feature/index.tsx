import { Container, Title, SubTitle } from "./styles/feature";

interface Props {}

interface TitleProps {}
interface SubTitleProps {}

interface Compound {
  Title: React.FC<TitleProps>;
  SubTitle: React.FC<SubTitleProps>;
}

const Feature: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const FTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const FSubTitle: React.FC<SubTitleProps> = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Feature.Title = FTitle;
Feature.SubTitle = FSubTitle;

export { Feature };
