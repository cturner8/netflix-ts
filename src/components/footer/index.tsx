import {
  Container,
  Row,
  Column,
  Title,
  Link,
  Text,
  Break,
} from "./styles/footer";

export interface Props {}

interface RowProps {}
interface ColumnProps {}
interface TitleProps {}
interface LinkProps {
  href: string;
}
interface TextProps {}
interface BreakProps {}

interface Compound {
  Row: React.FC<RowProps>;
  Column: React.FC<ColumnProps>;
  Title: React.FC<TitleProps>;
  Link: React.FC<LinkProps>;
  Text: React.FC<TextProps>;
  Break: React.FC<BreakProps>;
}

const Footer: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const FRow: React.FC<RowProps> = ({ children, ...restProps }) => {
  return <Row {...restProps}>{children}</Row>;
};

const FColumn: React.FC<ColumnProps> = ({ children, ...restProps }) => {
  return <Column {...restProps}>{children}</Column>;
};

const FLink: React.FC<LinkProps> = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

const FTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const FText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const FBreak: React.FC<BreakProps> = ({ children, ...restProps }) => {
  return <Break {...restProps}>{children}</Break>;
};

Footer.Row = FRow;
Footer.Column = FColumn;
Footer.Title = FTitle;
Footer.Link = FLink;
Footer.Text = FText;
Footer.Break = FBreak;

export { Footer };
