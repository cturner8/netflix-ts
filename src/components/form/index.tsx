import {
  InputHTMLAttributes,
  FormHTMLAttributes,
  ButtonHTMLAttributes,
} from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

interface Props {}

interface ErrorProps {}
interface BaseProps extends FormHTMLAttributes<HTMLFormElement> {}
interface TitleProps {}
interface TextProps {}
interface TextSmallProps {}
interface LinkProps {
  to: string;
}
interface InputProps extends InputHTMLAttributes<HTMLInputElement> {}
interface SubmitProps extends ButtonHTMLAttributes<HTMLButtonElement> {}

interface Compound {
  Error: React.FC<ErrorProps>;
  Base: React.FC<BaseProps>;
  Title: React.FC<TitleProps>;
  Text: React.FC<TextProps>;
  TextSmall: React.FC<TextSmallProps>;
  Link: React.FC<LinkProps>;
  Input: React.FC<InputProps>;
  Submit: React.FC<SubmitProps>;
}

const Form: React.FC<Props> & Compound = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const FError: React.FC<ErrorProps> = ({ children, ...restProps }) => {
  return <Error {...restProps}>{children}</Error>;
};

const FBase: React.FC<BaseProps> = ({ children, ...restProps }) => {
  return <Base {...restProps}>{children}</Base>;
};

const FTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const FText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const FTextSmall: React.FC<TextSmallProps> = ({ children, ...restProps }) => {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

const FLink: React.FC<LinkProps> = ({ children, to, ...restProps }) => {
  return (
    <Link to={to} {...restProps}>
      {children}
    </Link>
  );
};

const FInput: React.FC<InputProps> = ({ children, ...restProps }) => {
  return <Input {...restProps}>{children}</Input>;
};

const FSubmit: React.FC<SubmitProps> = ({ children, ...restProps }) => {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Error = FError;
Form.Base = FBase;
Form.Title = FTitle;
Form.Text = FText;
Form.TextSmall = FTextSmall;
Form.Link = FLink;
Form.Input = FInput;
Form.Submit = FSubmit;

export { Form };
