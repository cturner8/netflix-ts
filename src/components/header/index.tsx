import {
  ImgHTMLAttributes,
  HTMLAttributes,
  useState,
  Dispatch,
  SetStateAction,
  InputHTMLAttributes,
} from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureCallOut,
  Link,
  Group,
  Picture,
  Dropdown,
  Profile,
  Search,
  SearchInput,
  SearchIcon,
  PlayButton,
} from "./styles/header";

export interface Props {
  bg?: boolean;
  src?: string;
  dontShowSmallViewPort?: boolean;
}

interface FrameProps {}
interface GroupProps {}
export interface PictureProps {
  src: string | null | undefined;
}
interface ProfileProps {}
interface LogoProps extends ImgHTMLAttributes<HTMLImageElement> {
  to: string;
}
interface ButtonLinkProps {
  to: string;
}

interface DropdownProps {}
interface FeatureProps {}
interface TextProps {}
export interface TextLinkProps extends HTMLAttributes<HTMLParagraphElement> {
  active: "true" | "false";
}

interface FeatureCallOutProps {}

type SearchTerm = string;
type SetSearchTerm = Dispatch<SetStateAction<SearchTerm>>;
interface SearchTermProps {
  searchTerm: SearchTerm;
  setSearchTerm: SetSearchTerm;
}

export interface SearchInputProps
  extends InputHTMLAttributes<HTMLInputElement> {
  active: boolean;
}

interface PlayButtonProps {}

interface Compound {
  Frame: React.FC<FrameProps>;
  Group: React.FC<GroupProps>;
  Logo: React.FC<LogoProps>;
  ButtonLink: React.FC<ButtonLinkProps>;
  Feature: React.FC<FeatureProps>;
  Text: React.FC<TextProps>;
  TextLink: React.FC<TextLinkProps>;
  FeatureCallOut: React.FC<FeatureCallOutProps>;
  Picture: React.FC<PictureProps>;
  Profile: React.FC<ProfileProps>;
  Dropdown: React.FC<DropdownProps>;
  SearchTerm: React.FC<SearchTermProps>;
  PlayButton: React.FC<PlayButtonProps>;
}

const Header: React.FC<Props> & Compound = ({
  bg = true,
  children,
  ...restProps
}) => {
  return bg ? (
    <Background {...restProps}>{children}</Background>
  ) : (
    <>{children}</>
  );
};

const HFrame: React.FC<FrameProps> = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const HGroup: React.FC<GroupProps> = ({ children, ...restProps }) => {
  return <Group {...restProps}>{children}</Group>;
};

const HPicture: React.FC<PictureProps> = ({ src, ...restProps }) => {
  return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

const HProfile: React.FC<ProfileProps> = ({ children, ...restProps }) => {
  return <Profile {...restProps}>{children}</Profile>;
};

const HSearchTerm: React.FC<SearchTermProps> = ({
  searchTerm,
  setSearchTerm,
  ...restProps
}) => {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setSearchActive((prevActive) => !prevActive)}>
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series"
        active={searchActive}
      />
    </Search>
  );
};

const HLogo: React.FC<LogoProps> = ({ to, ...restProps }) => {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

const HButtonLink: React.FC<ButtonLinkProps> = ({ ...restProps }) => {
  return <ButtonLink {...restProps} />;
};

const HFeatureCallOut: React.FC<FeatureCallOutProps> = ({
  children,
  ...restProps
}) => {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

const HFeature: React.FC<FeatureProps> = ({ children, ...restProps }) => {
  return <Feature {...restProps}>{children}</Feature>;
};

const HDropdown: React.FC<DropdownProps> = ({ children, ...restProps }) => {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

const HText: React.FC<TextProps> = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

const HTextLink: React.FC<TextLinkProps> = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

const HPlayButton: React.FC<PlayButtonProps> = ({ children, ...restProps }) => {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Frame = HFrame;
Header.Group = HGroup;
Header.Logo = HLogo;
Header.ButtonLink = HButtonLink;
Header.Feature = HFeature;
Header.Text = HText;
Header.TextLink = HTextLink;
Header.FeatureCallOut = HFeatureCallOut;
Header.Picture = HPicture;
Header.Profile = HProfile;
Header.Dropdown = HDropdown;
Header.SearchTerm = HSearchTerm;
Header.PlayButton = HPlayButton;

export { Header };
