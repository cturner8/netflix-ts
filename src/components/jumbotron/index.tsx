import {
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
  Item,
} from "./styles/jumbotron";

export interface Props {
  direction?: string;
}

interface ContainerProps {}
interface PaneProps {}
interface TitleProps {}
interface SubTitleProps {}
interface ImageProps {
  src: string;
  alt: string;
}

interface Compound {
  Container: React.FC<ContainerProps>;
  Pane: React.FC<PaneProps>;
  Title: React.FC<TitleProps>;
  SubTitle: React.FC<SubTitleProps>;
  Image: React.FC<ImageProps>;
}

const Jumbotron: React.FC<Props> & Compound = ({
  children,
  direction = "row",
  ...restProps
}) => {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
};

const JContainer: React.FC<ContainerProps> = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

const JPane: React.FC<PaneProps> = ({ children, ...restProps }) => {
  return <Pane {...restProps}>{children}</Pane>;
};

const JTitle: React.FC<TitleProps> = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

const JSubTitle: React.FC<SubTitleProps> = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

const JImage: React.FC<ImageProps> = ({ ...restProps }) => {
  return <Image {...restProps} />;
};

Jumbotron.Container = JContainer;
Jumbotron.Pane = JPane;
Jumbotron.Title = JTitle;
Jumbotron.SubTitle = JSubTitle;
Jumbotron.Image = JImage;

export { Jumbotron };
