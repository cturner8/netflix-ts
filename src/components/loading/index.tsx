import { Spinner, LockBody, Picture, ReleaseBody } from "./styles/index";

interface Props {
  src: string | null | undefined;
}

interface ReleaseBodyProps {}

interface Compound {
  ReleaseBody: React.FC<ReleaseBodyProps>;
}

const Loading: React.FC<Props> & Compound = ({ src, ...restProps }) => {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`images/users/${src}.png`} />
    </Spinner>
  );
};

const LReleaseBody: React.FC<ReleaseBodyProps> = () => {
  return <ReleaseBody />;
};

Loading.ReleaseBody = LReleaseBody;

export { Loading };
