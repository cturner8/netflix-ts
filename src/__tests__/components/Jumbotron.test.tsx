import { render, RenderResult } from "@testing-library/react";
import { Jumbotron } from "../../components";
import jumboData from "../../fixtures/jumbo.json";

describe("<Jumbotron />", () => {
  let component: RenderResult;

  beforeEach(() => {
    component = render(
      <Jumbotron.Container>
        {jumboData.map((item) => (
          <Jumbotron key={item.id}>
            <Jumbotron.Pane>
              <Jumbotron.Title>{item.title}</Jumbotron.Title>
              <Jumbotron.SubTitle>{item.subTitle}</Jumbotron.SubTitle>
            </Jumbotron.Pane>
            <Jumbotron.Pane>
              <Jumbotron.Image
                src={item.image}
                alt={item.alt}
                data-testid={`${item.id}-jumbo-image`}
              />
            </Jumbotron.Pane>
          </Jumbotron>
        ))}
      </Jumbotron.Container>
    );
  });

  it("renders the jumbotron with populated data", () => {
    const { container, getByText, getByAltText, getByTestId } = component;

    expect(getByText(/Enjoy on your TV./i)).toBeTruthy();
    expect(getByAltText(/Tiger King on Netflix/i)).toBeTruthy();
    expect(getByTestId("1-jumbo-image")).toBeTruthy();
    expect(
      getByText(
        /Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more./i
      )
    ).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
