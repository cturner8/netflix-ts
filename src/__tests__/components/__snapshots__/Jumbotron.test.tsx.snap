// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders the jumbotron with populated data 1`] = `
<div
  class="jumbotron__Container-sc-27ubqc-2 cykytk"
>
  <div
    class="jumbotron__Item-sc-27ubqc-0 dmLimn"
  >
    <div
      class="jumbotron__Inner-sc-27ubqc-1 eFoPXa"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-27ubqc-3 buCGkg"
      >
        <h1
          class="jumbotron__Title-sc-27ubqc-4 ixzRWN"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-27ubqc-5 kuDOJe"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-27ubqc-3 buCGkg"
      >
        <img
          alt="Tiger King on Netflix"
          class="jumbotron__Image-sc-27ubqc-6 kdCWJW"
          data-testid="1-jumbo-image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-27ubqc-0 dmLimn"
  >
    <div
      class="jumbotron__Inner-sc-27ubqc-1 eFoPXa"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-27ubqc-3 buCGkg"
      >
        <h1
          class="jumbotron__Title-sc-27ubqc-4 ixzRWN"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-27ubqc-5 kuDOJe"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-27ubqc-3 buCGkg"
      >
        <img
          alt="Watch on mobile"
          class="jumbotron__Image-sc-27ubqc-6 kdCWJW"
          data-testid="2-jumbo-image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="jumbotron__Item-sc-27ubqc-0 dmLimn"
  >
    <div
      class="jumbotron__Inner-sc-27ubqc-1 eFoPXa"
      direction="row"
    >
      <div
        class="jumbotron__Pane-sc-27ubqc-3 buCGkg"
      >
        <h1
          class="jumbotron__Title-sc-27ubqc-4 ixzRWN"
        >
          Watch everywhere.
        </h1>
        <h2
          class="jumbotron__SubTitle-sc-27ubqc-5 kuDOJe"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="jumbotron__Pane-sc-27ubqc-3 buCGkg"
      >
        <img
          alt="Money Heist on Netflix"
          class="jumbotron__Image-sc-27ubqc-6 kdCWJW"
          data-testid="3-jumbo-image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
